### PR TITLE
implemented getting user by id

### PR DESCRIPTION
**Pull request by Odelola**
In this PR, retrieving users by their Id has been implemented.

In the `UserController` file the `getUserById` function find the PathVariable in the url and then passes it to the `getUserById` function coming from userService and then it has to be parsed to be a Dto from a model.

In the `UserService` file it is just the declaration of the function that is there.

In the `UserServiceImpl` file the `getUserById` function calls the `findUserById` coming from the `UserRepository` file and then parses it to a Model from an Entity because that is what is returned from the function from the UserRepository.

In the `UserRepository` file, I map over the users "database" that have been implemented in the `findAllUsers` function. Retrieve it and the stream and filter over it to get the user object that has its id matched to the userId in the arguments list.

This really caused a lot of time because I was researching on how it was done I saw people implement it but at the same time I also saw many people do it without the repositories, that made me try to understand it a bit.. When I understood that repository is acting like our database that actually gives us something then I had to come up with a way for it to give me the User I was looking for and I also struggled with syntaxes of functions declarations.

### CODE REVIEW

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 spring.application.name=hello
-spring.datasource.driver-class-name=
-spring.datasource.url=
-spring.datasource.username=
-spring.datasource.password=
\ No newline at end of file
+spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
```
Comment (dejiadeloye) [inline src/main/resources/application.properties]: Never commit your credentials

File: src/main/java/com/example/hello/users/service/UserServiceImpl.java [hunk 2]
```diff
@@ -19,6 +20,11 @@ public class UserServiceImpl implements UserService {
 
     private final UserMapper userMapper;
 
+
+    //private final UserRepository userMapper2;
+    private final UserRepository userRepository2;
```
Comment (dejiadeloye) [inline src/main/java/com/example/hello/users/service/UserServiceImpl.java]: why userRepository2? There is no userRepository1?, Rename this to a proper name

File: src/main/java/com/example/hello/users/service/UserServiceImpl.java [hunk 3]
```diff
@@ -30,6 +36,15 @@ public List<UserModel> getUsers() {
     @Override
     public UserModel getUserById(UUID userId) {
         // TODO Auto-generated method stub
-        throw new UnsupportedOperationException("Unimplemented method 'getUserById'");
+        //var userRepository = this.getUsers();
+
```
Comment (dejiadeloye) [inline src/main/java/com/example/hello/users/service/UserServiceImpl.java]: Remove comments

File: src/main/java/com/example/hello/users/repository/UserRepository.java [hunk 4]
```diff
@@ -38,7 +39,18 @@ public List<UserEntity> findAllUsers() {
         return users;
     }
 
+    //private final UserRepository userRepository;
```
Comment (dejiadeloye) [inline src/main/java/com/example/hello/users/repository/UserRepository.java]: Remove comments

File: src/main/java/com/example/hello/users/controller/UserController.java [hunk 5]
```diff
@@ -1,14 +1,21 @@
...
+
 import java.util.List;
+import java.util.UUID;
 
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.PathVariable;
+
```
Comment (dejiadeloye) [inline src/main/java/com/example/hello/users/controller/UserController.java:11]: remove unnecessary new lines

File: src/main/java/com/example/hello/users/service/UserServiceImpl.java (hunk 2 above)
Comment (Odelola) [reply src/main/java/com/example/hello/users/service/UserServiceImpl.java]: I have removed the variable.

File: src/main/java/com/example/hello/users/repository/UserRepository.java (hunk 4 above)
Comment (Odelola) [reply src/main/java/com/example/hello/users/repository/UserRepository.java]: I have removed all comments.

File: src/main/java/com/example/hello/users/controller/UserController.java (hunk 5 above)
Comment (Odelola) [reply src/main/java/com/example/hello/users/controller/UserController.java:11]: I have removed all unnecessary lines

File: src/main/java/com/example/hello/users/service/UserServiceImpl.java (hunk 3 above)
Comment (Odelola) [reply src/main/java/com/example/hello/users/service/UserServiceImpl.java]: I have removed all comments

File: src/main/resources/application.properties (hunk 1 above)
Comment (Odelola) [reply src/main/resources/application.properties]: I have removed the credentials I would pay attention to this mistake.

**Review by dejiadeloye — APPROVED** [review]

**Review by adeloyedeji — APPROVED** [review]

**Pull request merged**